architecture/core_concepts/templates.adoc

```diff
@@ -10,63 +10,133 @@
-customized and processed to produce a configuration. Each template defines a
-list of parameters that can be modified for consumption by containers.
+[[overview]]
 
-Administrators and developers can link:../../dev_guide/templates.html[interact
-with templates] using the CLI and web console.
+== Overview
+A template describes a set of link:overview.html[objects] that
+can be parameterized and processed to produce a list of objects
+for creation by OpenShift. The objects to create can include
+anything that users have permission to create within a project,
+for example link:pods_and_services.html#services[services],
+link:builds_and_image_streams.html#builds[build configurations], and
+link:deployments.html#deployments-and-deployment-configurations[deployment
+configurations]. A template may also define a set of
+link:pods_and_services.html#labels[labels] to apply to every object
+defined in the template.
 
-.Template Object Definition
+.A Simple Template Object Definition (YAML)
 ====
 
-[source,json]
+[source,yaml]
 ----
-{
-  "kind": "Template",
-  "apiVersion": "v1",
-  "objects": [
-    {
-      "kind": "Pod",
-      "apiVersion": "v1",
-      "metadata": {
-        "name": "redis-master",
-        "labels": {
-          "name": "redis-master"
-        }
-      },
-      "spec": {
-        "containers": [
-          {
-            "name": "master",
-            "image": "dockerfile/redis",
-            "ports": [
-              {
-                "containerPort": 6379,
-                "protocol": "TCP"
-              }
-            ],
-            "env": [
-              {
-                "name": "REDIS_PASSWORD",
-                "value": "${REDIS_PASSWORD}"
-              }
-            ]
-          }
-        ]
-      }
-    }
-  ],
-  "parameters": [
-    {
-      "name": "REDIS_PASSWORD",
-      "description": "Password used for Redis authentication",
-      "generate": "expression",
-      "from": "[A-Z0-9]{8}"
-    }
-  ]
-}
+apiVersion: v1
+kind: Template
+metadata:
+  name: redis-template <1>
+  annotations:
+    description: "Description" <2>
+    iconClass: "icon-redis" <3>
+    tags: "database,nosql" <4>
+objects:   <5>
+- apiVersion: v1
+  kind: Pod
+  metadata:
+    name: redis-master
+  spec:
+    containers:
+    - env:
+      - name: REDIS_PASSWORD
+        value: ${REDIS_PASSWORD} <6>
+      image: dockerfile/redis
+      name: master
+      ports:
+      - containerPort: 6379
+        protocol: TCP
+parameters:  <7>
+- description: Password used for Redis authentication
+  from: '[A-Z0-9]{8}'   <8>
+  generate: expression
+  name: REDIS_PASSWORD
+labels:      <9>
+  redis: master
 ----
 
+<1> The name of the template
+<2> Optional description for the template
+<3> The icon that will be shown in the UI for this template; the name of a CSS class
+defined in the link:https://raw.githubusercontent.com/openshift/origin/master/pkg/assets/bindata.go[the web console source] (search for "openshift-logos-icon").
```

Comment:
This can probably do for now, and we could doc the list somewhere soon. Not super sure about best place to do so; maybe the web console arch topic? Maybe as an appendix?

Also, there's a double "the" in this sentence.